bin/boot

```diff
@@ -26,49 +26,53 @@ function login  {
     set -e
 }
 
-function _update_odoo_conf_param {
-    if [ "$2" ]; then
-        echo $log_src[`date +%F.%H:%M:%S`]' Replacing parameter '$1
-        sudo -i -u \#$TARGET_ID sed -i "s/$1/$2/g" $odoo_conf_file
-    fi
+function _update_odoo_conf_params {
+    # Loop over all the "ODOO_" ENV variables
+    while read -r var; do
+        # Remove "ODOO_" from ENV variable and convert to lowercase
+        param=${var:5}
+        param=${param,,}
+
+        # Get the value of the corresponding ENV variable and escape slashes
+        val=${!var}
+        val=$( echo $val | sed 's/\//\\\//g')
+
+        bash /app/bin/update_odoo_param.sh $odoo_user $odoo_conf_file $param $val
+
+        # Unset the environment variable for security purpose
+        unset $var
+    done <<< "$( printenv | grep '^ODOO_' | sed 's/=.*//g' )"
 }
 
-function start {
-    echo $log_src[`date +%F.%H:%M:%S`]' Checking special requirements...'
-    bash /opt/scripts/startup.sh
-
-    echo $log_src[`date +%F.%H:%M:%S`]' Updating Odoo conf...'
-    _update_odoo_conf_param '_ADMIN_PASSWORD_' $ADMIN_PASSWORD
-    _update_odoo_conf_param '_DB_USER_' $DB_USER
-    _update_odoo_conf_param '_DB_PASSWORD_' $DB_PASSWORD
-    _update_odoo_conf_param '_DB_FILTER_' $DB_FILTER
-    echo $log_src[`date +%F.%H:%M:%S`]' Update Odoo conf done'
+function _setup_ssh_key {
+    echo $log_src[`date +%F.%H:%M:%S`]' Setting up SSH key...'
+    ssh_folder=$( getent passwd $odoo_user | cut -d: -f6 )/.ssh
+    echo $log_src[`date +%F.%H:%M:%S`]' Creating SSH folder = '$ssh_folder
+    mkdir $ssh_folder
+    echo $log_src[`date +%F.%H:%M:%S`]' Copy SSH private key from /opt/odoo/ssh'
+    cp /opt/odoo/ssh/id_rsa $ssh_folder
+    echo $log_src[`date +%F.%H:%M:%S`]' Scanning GitHub key...'
+    ssh-keyscan github.com > $ssh_folder/known_hosts
+    echo $log_src[`date +%F.%H:%M:%S`]' Setup SSH config...'
+    echo "host github.com" > $ssh_folder/config
+    echo "    HostName github.com" >> $ssh_folder/config
+    echo "    User git" >> $ssh_folder/config
+    echo "    IdentityFile $ssh_folder/id_rsa" >> $ssh_folder/config
+    echo $log_src[`date +%F.%H:%M:%S`]' Securing SSH folder...'
+    chown -R $odoo_user:$odoo_user $ssh_folder
+    chmod 400 $ssh_folder/id_rsa
+    echo $log_src[`date +%F.%H:%M:%S`]' Setup SSH key done'
+}
 
-    echo $log_src[`date +%F.%H:%M:%S`]' Check if need to download addons repo...'
+function _download_addons {
     if [ "$ADDONS_REPO" -o -a /opt/odoo/additional_addons/oca_dependencies.txt ]; then
-        echo $log_src[`date +%F.%H:%M:%S`]' Git config for target user'
-        sudo -i -u \#$TARGET_ID git config --global user.email "contact@elico-corp.com"
-        sudo -i -u \#$TARGET_ID git config --global user.name "Elico Corp - Odoo Docker"
+        # Git config for target user
+        sudo -i -u $odoo_user git config --global user.email "contact@elico-corp.com"
```

Comment:
Anyone using our public image will actually use that setup, this is why I preferred to use our business contact (our infra team doesn't care that Company X has pulled some code using our Odoo Docker image)